storage/src/checker/mod.rs

```diff
@@ -0,0 +1,69 @@
+// Copyright (C) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE.md for licensing terms.
+
+mod error;
+mod range_set;
+
+use crate::{
+    Child, Committed, FileBacked, HashedNodeReader, LinearAddress, Node, NodeReader, NodeStore,
+};
+use range_set::LinearAddressRangeSet;
+
+pub use error::CheckerError;
+
+/// Go through the filebacked storage and check for any inconsistencies. It proceeds in the following steps:
+/// 1. traverse the trie and check the nodes
+/// 2. check the free list
+/// 3. check any bubbles - what are the spaces between trie nodes and free lists?
+// TODO: add merkle hash checks as well
+pub async fn check_node_store(
+    node_store: &NodeStore<Committed, FileBacked>,
+) -> Result<(), CheckerError> {
+    let mut visited = LinearAddressRangeSet::new(node_store.storage.get_file_size()?)?;
+
+    // 1. traverse the trie and check the nodes
+    let root_address = node_store.root_address_and_hash()?.map(|(addr, _)| addr);
```

Comment:
Not sure I love getting the address and hash here, since the implementation reads the root node, and then you read it again. Options:
 - Maybe pass a `Option<SharedNode>` down instead. You can get the root one from the `root_node` method on the `NodeStore`.
 - Maybe create a `root_address` method next to `root_address_and_hash` which just returns the root address.

Reply:
The problem only exists if we need to access the address from an external file: when we move everything into `nodestore.rs`, we have direct access to `node_store.header.root`.